File: sem_4/www/lab4/models/Event.js
const mongoose = require('mongoose');

const eventSchema = new mongoose.Schema({
  title: {
    type: String,
    required: [true, 'Event title is required'],
    trim: true
  },
  description: {
    type: String,
    required: [true, 'Event description is required'],
    trim: true
  },
  date: {
    type: Date,
    required: [true, 'Event date is required'],
    validate: {
      validator: (value) => !isNaN(Date.parse(value)),
      message: 'Invalid date format'
    }
  },
  creator: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: [true, 'Creator is required']
  }
});

module.exports = mongoose.model('Event', eventSchema);

File: sem_4/www/lab5_6/event_app/lib/models/event.dart
import 'package:event_app/models/user.dart';

class Event {
  final String id;
  final String title;
  final String description;
  final DateTime date;
  final String creatorId;
  final User? creator;

  Event({
    required this.id,
    required this.title,
    required this.description,
    required this.date,
    required this.creatorId,
    this.creator,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['_id'],
      title: json['title'],
      description: json['description'],
      date: DateTime.parse(json['date']),
      creatorId: json['creator'] is Map ? json['creator']['_id'] : json['creator'],
      creator: json['creator'] is Map ? User.fromJson(json['creator']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'description': description,
      'date': date.toIso8601String(),
      'creator': creatorId,
    };
  }
}